cf/forest-snapshot-listing/src/worker.ts

```diff
@@ -0,0 +1,100 @@
+const BASE_URL = 'https://forest-archive.chainsafe.dev';
+
+interface Env {
+  FOREST_ARCHIVE: R2Bucket;
+}
+
+async function do_listing(env: Env, prefix) {
+  const options = {
+    limit: 500,
+    prefix: prefix,
+  };
+
+  const listed = await env.FOREST_ARCHIVE.list(options);
+  let truncated = listed.truncated;
+  let cursor = listed.truncated ? listed.cursor : undefined;
+
+  // use the truncated property to check if there are more
+  // objects to be returned
+  while (truncated) {
+    const next = await env.FOREST_ARCHIVE.list({
+      ...options,
+      cursor: cursor,
+    });
+    listed.objects.push(...next.objects);
+
+    truncated = next.truncated;
+    cursor = next.truncated ? next.cursor : undefined;
+  }
+
+  let html = `<!DOCTYPE html>
+		<body>
+		  <h1>Forest Archive</h1>
+      <ul>`;
+  for (const obj of listed.objects) {
+    html += `<li><a href="${BASE_URL}/${obj.key}">${obj.key}</a></li>\n`;
+  }
+  html += `</ul></body>`;
+
+  return new Response(html, {
+    headers: {
+      'content-type': 'text/html;charset=UTF-8',
+    },
+  });
+}
+
+export default {
```

Comment:
nit: I would rather use a template as opposed to generate html within the controller code, but if we aren't going to touch this much - I guess it does not make a whole lot of difference.